checks/evaluation/pinned_dependencies.go

```diff
@@ -90,66 +95,32 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 				Snippet:     rr.Location.Snippet,
 				Remediation: generateRemediation(remediationMetadata, &rr),
 			})
-
-			// Update the pinning status.
-			updatePinningResults(&rr, &wp, pr)
 		}
+		// Update the pinning status.
+		updatePinningResults(&rr, &wp, pr)
 	}
 
 	// Generate scores and Info results.
-	// GitHub actions.
-	actionScore, err := createReturnForIsGitHubActionsWorkflowPinned(wp, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Docker files.
-	dockerFromScore, err := createReturnForIsDockerfilePinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Docker downloads.
-	dockerDownloadScore, err := createReturnForIsDockerfileFreeOfInsecureDownloads(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Script downloads.
-	scriptScore, err := createReturnForIsShellScriptFreeOfInsecureDownloads(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Pip installs.
-	pipScore, err := createReturnForIsPipInstallPinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
-	}
-
-	// Npm installs.
-	npmScore, err := createReturnForIsNpmInstallPinned(pr, dl)
-	if err != nil {
-		return checker.CreateRuntimeErrorResult(name, err)
+	var scores []*checker.ProportionalScoreWithWeight
+	// Go through all dependency types
+	// GitHub Actions need to be handled separately since they are not in pr
+	scores = append(scores, createScoreForGitHubActionsWorkflow(&wp)...)
+	// Only exisiting dependencies will be found in pr
+	// We will only score the ecossystem if there are dependencies
+	// This results in only existing ecossystems being included in the final score
+	for t := range pr {
+		scores = append(scores, &checker.ProportionalScoreWithWeight{
+			Success: pr[t].pinned,
+			Total:   pr[t].total,
+			Weight:  10,
```

Comment:
can we make this weight a `const` (perhaps right before this loop?), with a comment about why 10? I assume it's because github actions are split 2 and 8?

Reply:
if we make 2 and 8 consts, we can also define this const in terms of those const, that way we ensure GitHub action will always be weighted equally to other types

Reply:
Although thinking about this now, shouldn't the weight be dependent on total? If all ecosystems/types have weight 10, then we're still averaging the ecosystems unweighted.

Reply:
> Although thinking about this now, shouldn't the weight be dependent on total?

No, I don't think so. 

> If all ecosystems/types have weight 10, then we're still averaging the ecosystems unweighted.

Yes, that's true! The only reason for the weighting is to make GHA GitHub-owned and third-party have different weights.

e.g. 1 / 20 GHA GitHub-owned + 2 / 20 GHA third-party + 1 / 1 pipCommand
```
(1 * 2) + (2 * 8) + (1 * 10)
-------------------------------
(20 * 2) + (20 * 8) + (1 * 10)

= 28 / 210 = 0.13
```

e.g. 1 / 20 GHA GitHub-owned + 20 / 20 GHA third-party + 1 / 1 pipCommand
```
(1 * 2) + (20 * 8) + (1 * 10)
-------------------------------
(20 * 2) + (20 * 8) + (1 * 10)

= 172 / 210 = 0.81
```

e.g. 20 / 20 GHA GitHub-owned + 2 / 20 GHA third-party + 1 / 1 pipCommand
```
(20 * 2) + (2 * 8) + (1 * 10)
-------------------------------
(20 * 2) + (20 * 8) + (1 * 10)

= 66 / 210 = 0.31
```

This makes Third-party count more than GitHub-owned.

Reply:
Yes, I think I was confused about where the weights was being taken, so ignore that part. Please take a look at this part still though:
> if we make 2 and 8 consts, we can also define this const in terms of those const, that way we ensure GitHub action will always be weighted equally to other types

something like
```go
const (
    githubOwnedActionWeight = 2
    thirdPartyActionWeight = 8
    nonActionWeight = githubOwnedActionWeight + thirdPartyActionWeight
)
```

Reply:
Yes! We can do that.